### PR TITLE
Break main() into several functions

### PR DESCRIPTION
**Pull request by marauder37**
This pull request includes a

- [x] Bug fix
- [ ] New feature
- [ ] Translation

The following changes were made

-Refactor main() into several functions
-Should be no changes to functionality, only factoring

If this is related to an existing ticket, include a link to it as well.

### CODE REVIEW

**Comment by nborrmann** [comment]
long overdue, but lots of conflicts with #101

**Comment by marauder37** [comment]
I'm trying to rebase it now

**Comment by AHAAAAAAA** [comment]
https://github.com/AHAAAAAAA/PokemonGo-Map/issues/58

**Comment by AHAAAAAAA** [comment]
Solves this issue #58

**Comment by marauder37** [comment]
Conflicts gone, can I get some super testing on this please?

File: example.py [hunk 1]
```diff
@@ -416,87 +410,106 @@ def main():
...
+        debug("looping: step {} of {}".format(step, steplimit ** 2))
+
+        if (-steplimit / 2 < x <= steplimit / 2) and (-steplimit / 2 < y <= steplimit / 2):
+            set_location_coords((x * 0.0025) + deflat, (y * 0.0025) + deflng, 0)
+        if x == y or (x < 0 and x == -y) or (x > 0 and x == 1 - y):
             dx, dy = -dy, dx
-        x, y = x+dx, y+dy
-        steps += 1
-        print("Completed:", ((steps + (pos * .25) - .25) / steplimit**2) * 100, "%")
+        x, y = x + dx, y + dy
```
Comment (AHAAAAAAA) [inline example.py]: We can probably delegate the location scan math to a function.

**Comment by sykhro** [comment]
Manually merged along other things, thanks for the contribution

**Pull request closed without merging**